core/tests/test_doubly_linked_list_uint8.h

```diff
@@ -0,0 +1,377 @@
+    doubly_linked_list_uint8_push_front(list, 5);
+    doubly_linked_list_uint8_push_front(list, 10);
+    doubly_linked_list_uint8_push_front(list, 15);
+    uint8_t valueCheck = 0;
+
+    TEST_CHECK(doubly_linked_list_uint8_pop_back(list, &valueCheck) == true);
+    TEST_CHECK(valueCheck == 5);
+    size_t checkSize = doubly_linked_list_uint8_node_count(list);
+    TEST_CHECK(checkSize == 2);
+    TEST_CHECK(doubly_linked_list_uint8_pop_back(list, &valueCheck) == true);
+    TEST_CHECK(valueCheck == 10);
+    checkSize = doubly_linked_list_uint8_node_count(list);
+    TEST_CHECK(checkSize == 1);
+    TEST_CHECK(doubly_linked_list_uint8_pop_back(list, &valueCheck) == true);
+    TEST_CHECK(valueCheck == 15);
+    checkSize = doubly_linked_list_uint8_node_count(list);
+    TEST_CHECK(checkSize == 0);
+    TEST_CHECK(list->front == NULL);
+    TEST_CHECK(list->back == NULL);
+
+    doubly_linked_list_uint8_free(list);
+}
+
+// Create a list with 2 nodes with differents values and then check if we get the good one with the
+// functions without taking it out of the list. Also test all of these function :
+// --- doubly_linked_list_uint8_front()
+// --- doubly_linked_list_uint8_back()
+//
+void test_doubly_linked_list_uint8_front_and_back(void) {
+    DoublyLinkedListUint8 *list = doubly_linked_list_uint8_new();
+    doubly_linked_list_uint8_push_front(list, 5);
+    doubly_linked_list_uint8_push_front(list, 10);
+
+    DoublyLinkedListUint8Node *nodeCheck = doubly_linked_list_uint8_front(list);
+    size_t check = doubly_linked_list_uint8_node_get_value(nodeCheck);
+    TEST_CHECK(check == 10);
+
+    nodeCheck = doubly_linked_list_uint8_back(list);
+    check = doubly_linked_list_uint8_node_get_value(nodeCheck);
+    TEST_CHECK(check == 5);
+
+    doubly_linked_list_uint8_free(list);
+}
+
+// Create a list with a node, then insert a node in the list before the first one. Then we do it a
+// second time. We check at each step if the values in the list are correct, and if the links are
+// not broken.
+void test_doubly_linked_list_uint8_insert_node_before(void) {
+    DoublyLinkedListUint8 *list = doubly_linked_list_uint8_new();
+    DoublyLinkedListUint8Node *aNode = doubly_linked_list_uint8_push_front(list, 5);
+
+    DoublyLinkedListUint8Node *bNode = doubly_linked_list_uint8_insert_node_before(list, aNode, 10);
+    size_t check = doubly_linked_list_uint8_node_get_value(list->front);
+    TEST_CHECK(check == 10);
+    check = doubly_linked_list_uint8_node_get_value(list->back);
+    TEST_CHECK(check == 5);
+    DoublyLinkedListUint8Node *NodeCheck = doubly_linked_list_uint8_node_previous(aNode);
+    check = doubly_linked_list_uint8_node_get_value(NodeCheck);
+    TEST_CHECK(check == 10);
+    NodeCheck = doubly_linked_list_uint8_node_next(bNode);
+    check = doubly_linked_list_uint8_node_get_value(NodeCheck);
+    TEST_CHECK(check == 5);
+
+    DoublyLinkedListUint8Node *cNode = doubly_linked_list_uint8_insert_node_before(list, aNode, 15);
```

Comment:
Maybe add a comment with the state of the list, like `[15, 10, 5]`, do it also everywhere you have 3 elements or more in a list